src/main/java/org/skriptlang/skript/bukkit/recipes/RecipeWrapper.java

```diff
@@ -0,0 +1,385 @@
+package org.skriptlang.skript.bukkit.recipes;
+
+import org.bukkit.NamespacedKey;
+import org.bukkit.inventory.*;
+import org.bukkit.inventory.recipe.CookingBookCategory;
+import org.bukkit.inventory.recipe.CraftingBookCategory;
+import org.skriptlang.skript.bukkit.recipes.RecipeUtils.RecipeType;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Wrapper class allowing the creation of a Recipe without being on initialization.
+ */
+public abstract class RecipeWrapper implements Recipe {
```

Comment:
This isn't a wrapper, it doesn't wrap a Recipe.
It's an implementation of Recipe that's mutable, hence it should really be MutableRecipe

Reply:
applies to all subclasses too